src/PerfView/Utilities/RangeUtilities.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Globalization;
+
+namespace Microsoft.Diagnostics.Utilities
+{
+    /// <summary>
+    /// Provides utility operations for <see cref="double"/> ranges.
+    /// </summary>
+    internal static class RangeUtilities
+    {
+        /// <summary>
+        /// Array of range separators built from the current culture.
+        /// </summary>
+        /// <remarks>The user can set any character as a separator and that character can turn a regular expression invalid.</remarks>
+        private static readonly string[] rangeSeparators = ResolveRangeSeparators();
+
+        /// <summary>
+        /// Resolves the range separators for the current culture.
+        /// </summary>
+        /// <returns>The range separators for the current culture.</returns>
+        private static string[] ResolveRangeSeparators()
+        {
+            var listSeparator = " ";
+
+            if (string.IsNullOrWhiteSpace(CultureInfo.CurrentCulture.NumberFormat.NumberGroupSeparator)
```

Comment:
Can you describe what the behavior here should be (e.g. how would you tell someone to use this feature)?  I don't think I'm following this.

Reply:
1. To determine the range separator, we start with ` `.
2. If the current culture's number group separator or number decimal separator is a white space, we change it to the list separator.
3. if the list separator is a white space or it matches the number group separator or number decimal separator, we change it to `|`.

I'll add this explanation to code and will gladly add to the docs, if you point me to where they are.

Reply:
Is there a reason to have #2 at all?  I wonder if we can't just do #1 and if that fails, go straight to #3.  It would make things simpler.

---
In reply to: [524062988](https://github.com/microsoft/perfview/pull/1288#discussion_r524062988) [](ancestors = 524062988)

Reply:
\#2 was my initial idea, but I found cases where it's still ambiguous. That's why I came up with \#3.

I'm OK with dropping \#2.

Reply:
Sounds good.  Go for it.

---
In reply to: [555350287](https://github.com/microsoft/perfview/pull/1288#discussion_r555350287) [](ancestors = 555350287)